core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/DefaultServiceExtensionContext.java

```diff
@@ -273,6 +233,19 @@ private void addDefaultExtensions(List<ServiceExtension> loadedExtensions) {
         baseDependencies.forEach(se -> loadedExtensions.add(0, se));
     }
 
+    private Config loadConfig() {
```

Comment:
I think the order of precedence should be config props < env props < system props, meaning system props overwrite env, and so on. I know that this probably wasn't the case before, but since you're touching it now... :)

Reply:
makes sense. done